### PR TITLE
ensure we rollout when image pull spec changes

### PR DESCRIPTION
**Pull request by deads2k**
See the comment in the diff.

/assign @bparees @mfojtik

### CODE REVIEW

**Comment by deads2k** [comment]
known flake

File: pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go [hunk 1]
```diff
@@ -285,6 +285,14 @@ func manageOpenShiftAPIServerDaemonSet_v311_00_to_latest(client appsclientv1.Dae
 	if len(imagePullSpec) > 0 {
 		required.Spec.Template.Spec.Containers[0].Image = imagePullSpec
 	}
+	// we set this so that when the requested image pull spec changes, we always have a diff.  Remember that we don't directly
+	// diff any fields on the daemonset because they can be rewritten by admission and we don't want to constantly be fighting
+	// against admission or defaults.  That was a problem with original versions of apply.
+	if required.Annotations == nil {
+		required.Annotations = map[string]string{}
+	}
+	required.Annotations["openshiftapiservers.operator.openshift.io/pull-spec"] = imagePullSpec
```
Comment (bparees) [inline pkg/operator/workloadcontroller/workload_controller_openshiftapiserver_v311_00.go]: i wonder if this shouldn't have been https://github.com/openshift/library-go/pull/256/files#diff-67e7aec66ff0f1fcd97e1340464bc7beR68 anyway

**Pull request merged**